components/stacks-network/src/chains_coordinator.rs

```diff
@@ -591,37 +609,63 @@ pub async fn publish_stacking_orders(
+                                ClarityName::try_from("version".to_owned()).unwrap(),
+                                ClarityValue::buff_from_byte(addr_version as u8),
+                            ),
+                            (
+                                ClarityName::try_from("hashbytes".to_owned()).unwrap(),
+                                ClarityValue::Sequence(SequenceData::Buffer(BuffData {
+                                    data: addr_bytes.as_bytes().to_vec(),
+                                })),
+                            ),
+                        ])
+                        .unwrap(),
+                    ),
+                    ClarityValue::UInt((bitcoin_block_height - 1).into()),
+                    ClarityValue::UInt(duration),
+                ];
+                if pox_version >= 4 {
+                    let signer_key = vec![i as u8; 33];
+                    arguments.push(ClarityValue::buff_from(signer_key).unwrap());
+                };
+
                 let stack_stx_tx = codec::build_contrat_call_transaction(
                     pox_contract_id,
                     "stack-stx".into(),
-                    vec![
-                        ClarityValue::UInt(stx_amount.into()),
-                        ClarityValue::Tuple(
-                            TupleData::from_data(vec![
-                                (
-                                    ClarityName::try_from("version".to_owned()).unwrap(),
-                                    ClarityValue::buff_from_byte(addr_version as u8),
-                                ),
-                                (
-                                    ClarityName::try_from("hashbytes".to_owned()).unwrap(),
-                                    ClarityValue::Sequence(SequenceData::Buffer(BuffData {
-                                        data: addr_bytes.as_bytes().to_vec(),
-                                    })),
-                                ),
-                            ])
-                            .unwrap(),
-                        ),
-                        ClarityValue::UInt((bitcoin_block_height - 1).into()),
-                        ClarityValue::UInt(duration),
-                    ],
+                    arguments,
                     nonce,
                     default_fee,
                     &hex_bytes(&account_secret_key).unwrap(),
                 );
-                let _ = stacks_rpc
-                    .post_transaction(&stack_stx_tx)
-                    .expect("Unable to broadcast transaction");
+                stacks_rpc.post_transaction(&stack_stx_tx)
             });
+
+        match stacking_result {
+            Ok(result) => {
+                if let Ok(result) = result.join() {
+                    match result {
+                        Ok(_) => {
+                            let _ = devnet_event_tx.send(DevnetEvent::success(format!(
+                                "stacking order for {} STX submitted",
```

Comment:
```suggestion
                                "Stacking order for {} STX submitted",
```
this stood out as lower case next to all of the other upper case logs.